sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,153 @@ class RotatingCertificateCredential implements TokenCredential {
+
+The second difference is here the application is preventing the credential from automatically initiating user interaction. Even though the application authenticates the user before the credential is used, further interaction might still be needed, for instance in the case that the user's refresh token expires, or a specific method require additional consent or authentication.
+
+By setting the option `disableAutomaticAuthentication` to `true` the credential will fail to automatically authenticate calls where user interaction is necessary. Instead, the credential will throw an error `AuthenticationRequiredError`. The following example demonstrates an application handling such an exception to prompt the user to authenticate only after some application logic has completed.
+
+```
+try
+{
+    await client.getSecret("secret-name");
+}
+catch (e)
+{
+  await ensureAnimationCompleted();
+
+  if(e instanceof AuthenticationRequiredError){
+    await credential.authenticate(e.scopes);
+    client.getSecret("secret-name");
+  }
+  else{
+    throw (e);
+  }
+}
+```
+
+### Persisting user authentication data
+
+Quite often applications desire the ability to be run multiple times without having to re-authenticate the user on each execution. This requires that data from credentials be persisted outside of the application memory so that it can authenticate silently on subsequent executions. Applications can persist this data using `tokenCachePersistenceOptions` when constructing the credential, and persisting the `authenticationRecord` returned from `authenticate`.
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
+
+To configure a credential, such as the `InteractiveBrowserCredential`, to persist token data, simply set the `tokenCachePersistenceOptions` option.
+
+```
+  const options: InteractiveBrowserCredentialOptions =  {
+    tokenCachePersistenceOptions: {
+      enabled: true
+    }
+  }
+  const credential = new InteractiveBrowserCredential(options);
+```
+
+#### Persisting the Authentication Record
+
+The `AuthenticationRecord` which is returned from the `authenticate`, contains data identifying an authenticated account. It is needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state.
+
+Here is an example of an application storing the `AuthenticationRecord` to the local file system after authenticating the user.
+
+```
+import fs from "fs";
+import { promisify } from "util";
+const AUTH_RECORD_PATH = "./tokencache.bin";
```

Comment:
Will this code work? I normally like to resolve the path before I send it over. If this works, then you can skip this comment (mark it as resolved).

Reply:
Same goes with all of the relative paths in the snippets of this PR!

Reply:
added the code to use the full path instead of relative path.